pkg/kubelet/dockershim/checkpoint_store.go

```diff
@@ -0,0 +1,85 @@
+package dockershim
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+)
+
+// CheckpointStore provides the interface for checkpoint storage backend
+type CheckpointStore interface {
+	Add(key string, data []byte) error
+	Get(key string) ([]byte, error)
+	Delete(key string) error
+	List() ([]string, error)
+}
+
+// FileStore is an implementation of CheckpointStore interface which stores checkpoint in file.
+type FileStore struct {
+	path string
+}
+
+func (fstore *FileStore) Add(key string, data []byte) error {
+	if _, err := os.Stat(fstore.path); os.IsNotExist(err) {
+		if err = os.MkdirAll(fstore.path, 755); err != nil {
+			return err
+		}
+	}
+	return ioutil.WriteFile(filepath.Join(fstore.path, key), data, 0644)
```

Comment:
As a point of consistency, you should always write to a tempfile and Rename() to the final name.  Otherwise it's possible for someone to read an invalid partial state.